### ISSUE TITLE
Add more instances for GEnum et al.

### ISSUE THREAD

**Issue by hvr** [dreixel/generic-deriving#24]
following a short conversation with @kosmikus I'm filing this issue to point out that some classes deserve more primitive instances defined.

E.g. for `GEnum` we currently have a rather arbitrary set of 3 instances predefined:

```haskell
instance GEnum Int	 
instance GEnum a => GEnum [a]	 
instance GEnum a => GEnum (Maybe a)	 
```

However, it makes sense (it made for me, and I had to define orphans in my code) to complete this with more instances for types from base (and at the very least for Haskell Report types) such as

 - All finite integral types (`Data.Word`, `Data.Int`, maybe FFI types)
 - `Ordering`
 - `Bool`
 - Tuples up to size 7 at least
 - `instance (GEnum a, GEnum b) => GEnum (Either a b)`
 - `Char`
 - ...

**Comment by dreixel** [comment]
I agree. Patches welcome! :-)

**Comment by RyanGlScott** [comment]
Whoops, I didn't mean to automatically close that.

(I meant to say that I'm working on a branch to do this, but mentioning things in commits is dangerous, it seems :) )

**Comment by dreixel** [comment]
Your commit happened to say "fix #24", so this happened :-)
